### PR TITLE
Show collection updates

### PR DESCRIPTION
**Pull request by courte**
- Moves the CollectionController to the metadata wrangler: NYPL-Simplified/metadata_wrangler#52
- Removes some of the more special characters from the `client_secret`, since they were getting in way of effortlessly `curl`ing the route
- Adds the column `last_checked` to collections, so we can:
- Return(s) works updated since a collection's last visit.
- Creates a datasource for a collection made during testing (like `Collection.register` does all the time)

Also: tests, removal of some commenty cruft, and the use of unicode encoding where it was immediately apparent.

Fixes NYPL-Simplified/metadata_wrangler#51.

### CODE REVIEW

**Comment by aslagle** [comment]
Looks good to me. It might also be useful to be able to pass in a date to use instead of last_checked.

**Comment by aslagle** [comment]
I also wonder if this will need to be paginated - it could be a lot of works if people don't check very often.

File: model.py [hunk 1]
```diff
@@ -6937,6 +6925,26 @@ def catalog_identifier(self, _db, identifier):
...
+        """Returns all of a collection's works that have been updated since the
+        last time the collection was checked"""
+        if self.catalog:
+            catalog_identifier_ids = [identifier.id for identifier in self.catalog]
+
+            # Find the works in the catalog that have been created or updated
+            # since the collection last checked.
+            query = _db.query(Work).join(Work.coverage_records)
+            query = query.join(Work.license_pools).join(Identifier)
+            query = query.filter(Identifier.id.in_(catalog_identifier_ids))
```
Comment (leonardr) [inline model.py]: There are going to be hundreds of thousands of Identifiers in a collection, right? Rather than iterate over all of them collecting their IDs, you should join Identifier against collections_identifiers against Collection and let the database do the work.

File: model.py [hunk 2]
```diff
@@ -6937,6 +6925,26 @@ def catalog_identifier(self, _db, identifier):
...
+            if self.last_checked:
+                query = query.filter(
+                    WorkCoverageRecord.timestamp > self.last_checked
+                )
+
+            return query.all()
```
Comment (leonardr) [inline model.py]: Have this return the query itself, rather than every item in the resultset. This will make it easy to use our existing pagination system.

**Comment by leonardr** [comment]
:+1:

**Comment by aslagle** [comment]
:+1:
